language_service/src/completion.rs

```diff
@@ -195,36 +209,47 @@ impl CompletionListBuilder {
                     self.current_sort_group, current_sort_prefix, name
                 ))
             },
+            detail: None,
         }));
 
         self.current_sort_group += 1;
     }
 
     /// Push a group of completions that are themselves sorted into subgroups
-    fn push_sorted_completions<'a, I>(&mut self, iter: I, kind: CompletionItemKind)
-    where
-        I: Iterator<Item = (&'a str, u32)>,
-    {
+    fn push_sorted_completions<'a>(
+        &mut self,
+        iter: impl Iterator<Item = (&'a str, String, u32)>,
+        kind: CompletionItemKind,
+    ) {
         self.items
-            .extend(iter.map(|(name, item_sort_group)| CompletionItem {
+            .extend(iter.map(|(name, detail, item_sort_group)| CompletionItem {
                 label: name.to_string(),
                 kind,
                 sort_text: Some(format!(
                     "{:02}{:02}{}",
                     self.current_sort_group, item_sort_group, name
                 )),
+                detail: if detail.is_empty() {
```

Comment:
Could `detail` really be an empty string (and even if it could, what's the harm in just letting it bubble up as a detail)? Feels like an unnecessary check.

Reply:
It seems like a code health thing to me. Currently `push_sorted_completions` only operates on the results from `get_callables` which are expected to always have non-empty details, but there is nothing in `push_sorted_completions` that enforces that, so have a simple `if` to handle that seems like it adds some safety to the code.
As for how that is handled by mapping empty strings to `None` that also seems like it could prevent sneaky bugs in the future, because logically an item should either have some detail or not have some detail. Allowing for a third state where it has detail but it is an empty string is unnecessary and will cause confusion.
That's my take on this. What are your thoughts?

Reply:
I've made the changes we discussed offline.